src/blockchain/bitcoin_rpc.rs

```diff
@@ -0,0 +1,392 @@
+                    if let Some((_, _)) =
+                        database.get_path_from_script_pubkey(&output.script_pubkey)?
+                    {
+                        debug!("{} output #{} is mine, adding utxo", tx.txid(), i);
+                        updates.set_utxo(&UTXO {
+                            outpoint: OutPoint::new(tx.txid(), i as u32),
+                            txout: output.clone(),
+                        })?;
+                        received += output.value;
+                    }
+                }
+
+                // Save the transaction
+                let details = TransactionDetails {
+                    transaction: Some(tx.clone()),
+                    txid: tx.txid(),
+                    height: Some(height as u32),
+                    received,
+                    sent,
+                    timestamp: 0,
+                };
+                debug!("Saving tx: {}", tx.clone().txid());
+                updates.set_tx(&details).unwrap();
+
+                // Mark that we've processed this txid
+                processed_txids.push(ltx.info.txid);
+            }
+
+            // Exit loop if no transactions remain, otherwise increment ltx_skip
+            if ltxs.len() < ltx_count {
+                break;
+            } else {
+                ltx_skip += ltx_count;
+            }
+        }
+
+        database.commit_batch(updates)?;
+        info!("Saved transactions");
+
+        Ok(())
+    }
+
+    async fn get_tx(&mut self, txid: &Txid) -> Result<Option<Transaction>, Error> {
+        let response = self.0.as_mut().unwrap().get_transaction(txid, Some(true))?;
+        let tx = response.transaction()?;
+        Ok(Some(tx))
+    }
+
+    async fn broadcast(&mut self, tx: &Transaction) -> Result<(), Error> {
+        self.0.as_mut().unwrap().send_raw_transaction(tx)?;
+        Ok(())
+    }
+
+    async fn get_height(&mut self) -> Result<usize, Error> {
+        let info = self.0.as_mut().unwrap().get_blockchain_info()?;
+        Ok(info.blocks as usize)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::descriptor::*;
+    use crate::{sled, Wallet};
+    use bitcoin::util::bip32::ExtendedPrivKey;
+    use bitcoin::{Amount, Network};
+    use bitcoincore_rpc::{Auth, Client};
+    use dirs::home_dir;
+
+    use std::str::FromStr;
+
+    use rand::distributions::Alphanumeric;
+    use rand::{thread_rng, Rng, RngCore};
+
+    fn rand_str() -> String {
+        thread_rng().sample_iter(&Alphanumeric).take(10).collect()
+    }
+
+    fn make_descriptors() -> (String, String) {
+        let mut seed = vec![0u8; 16];
+        thread_rng().fill_bytes(seed.as_mut_slice());
+
+        let network = Network::Bitcoin;
+        let sk = ExtendedPrivKey::new_master(network, &seed).unwrap();
+        let external = format!("wpkh({}/0/*)", sk.to_string());
+        let internal = format!("wpkh({}/1/*)", sk.to_string());
+        (external, internal)
```

Comment:
It's really slick how you can just include an xprv in a descriptor and the signer automatically knows how to sign. Nice work here!